dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SubSnpGenotypeReaderTest.java

```diff
@@ -0,0 +1,138 @@
+
+import javax.sql.DataSource;
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static uk.ac.ebi.eva.dbsnpimporter.test.TestUtils.assertContains;
+
+@RunWith(SpringRunner.class)
+@JdbcTest
+public class SubSnpGenotypeReaderTest extends ReaderTest {
+
+    private static final int PAGE_SIZE = 2000;
+
+    private static final int BATCH = 12068;
+
+    @Autowired
+    private DataSource dataSource;
+
+    private SubSnpGenotypeReader reader;
+
+    private List<SubSnpGenotype> expectedSubSnpGenotypes;
+
+    @Before
+    public void setUp() {
+        expectedSubSnpGenotypes = new ArrayList<>();
+
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505332, "C/C,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505333, "G/G,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505335, "A/A,C/C"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505336, "C/C,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505337, "G/G,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505339, "C/C,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505340, "C/C,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505342, "C/C,T/T"));
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12068, "12068",
+                26505343, "-/-,C/C"));
+    }
+
+    private SubSnpGenotypeReader buildReader(int batch, int pageSize) throws Exception {
+        SubSnpGenotypeReader fieldsReader = new SubSnpGenotypeReader(batch, dataSource, pageSize);
+        fieldsReader.afterPropertiesSet();
+        ExecutionContext executionContext = new ExecutionContext();
+        fieldsReader.open(executionContext);
+        return fieldsReader;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        reader.close();
+    }
+
+    @Test
+    public void testLoadData() throws Exception {
+        reader = buildReader(BATCH, PAGE_SIZE);
+        assertNotNull(reader);
+        assertEquals(PAGE_SIZE, reader.getPageSize());
+    }
+
+    @Test
+    public void testQuery() throws Exception {
+        reader = buildReader(BATCH, PAGE_SIZE);
+        List<SubSnpGenotype> readSnps = readAll(reader);
+
+        assertEquals(9, readSnps.size());
+        for (int i = 0; i < expectedSubSnpGenotypes.size() ; i++) {
+            assertEquals(readSnps.get(i), expectedSubSnpGenotypes.get(i));
+        }
+    }
+
+    @Test
+    public void testQueryWithDifferentBatch() throws Exception {
+
+        expectedSubSnpGenotypes = new ArrayList<>();
+        expectedSubSnpGenotypes.add(new SubSnpGenotype(12069, "12069",
```

Comment:
I'm re-reviewing this PR. I did a comment here, but I didn't mean to reuse the expectedSubSnpGenotypes, but to change this method or testQuery to have the same structure:

Option a) Populate completely the "expectedSubSnpGenotypes" list in the setup method, and filter it in "testQuery" and "testQueryWithDifferentBatch" when asserting the results.

Option b) Option a) + put together all the code in "testQueryWithDifferentBatch" inside "testQuery" (we are testing the same, querying all the variants in a given batch)

Option c) Remove setup method, and create a list of the expected data inside each test method.

Probably I like more option b). We are testing queries, but the code is exactly the same, so you can do it in a single test. If the code to test is different, then it makes sense use different tests (you would be testing different behavior).